hibernate-core/src/main/java/org/hibernate/SessionFactoryObserver.java

```diff
@@ -22,6 +22,14 @@
 	public void sessionFactoryCreated(SessionFactory factory);
 
 	/**
+	 * Callback to indicate that the given factory is about to be closed. This factory is open
+	 * when this method is called.
+	 *
+	 * @param factory The factory closed.
+	 */
+	public void sessionFactoryAboutToClose(SessionFactory factory);
```

Comment:
Most callback methods either use start/end or before/after prefixes/suffixes. I ran a search and this would be the first time we have a method containing the `AboutTo` keyword.

Because the `sessionFactoryCreated` cannot be changed anymore, we could rename the `sessionFactoryAboutToClose` method `sessionFactoryCloseStart`.

What di you think about it? Does it sound better to you too?